diff --git a/lua/plugins/telescope/init.lua b/lua/plugins/telescope/init.lua
index dca2d85..92b5b25 100644
--- a/lua/plugins/telescope/init.lua
+++ b/lua/plugins/telescope/init.lua
@@ -2,5 +2,5 @@ return {
   require("plugins.telescope.telescope"),
   require("plugins.telescope.undo"),
   require("plugins.telescope.adjacent"),
-  require("plugins.telescope.project"),
+  require("plugins.telescope.lsp-toggle"),
 }
diff --git a/lua/plugins/telescope/project.lua b/lua/plugins/telescope/project.lua
deleted file mode 100644
index 7269b3b..0000000
--- a/lua/plugins/telescope/project.lua
+++ /dev/null
@@ -1,20 +0,0 @@
-return {
-  "ahmedkhalf/project.nvim",
-  dependencies = { "telescope.nvim" },
-  event = "VeryLazy",
-  -- keys = { { "<leader>sp", require("telescope").extensions.projects.projects({}), desc = "Project Management" } },
-  config = function()
-    require("project_nvim").setup({
-      manual_mode = false,
-      detection_methods = { "pattern", "lsp" },
-      patterns = { "package.json", "index.norg", ".git" },
-      ignore_lsp = {},
-      exclude_dirs = { "/home/Vedant", "/home/vedant" },
-      show_hidden = true,
-      silent_chdir = true,
-      scope_chdir = "global",
-      datapath = vim.fn.stdpath("data"),
-    })
-    require("telescope").load_extension("projects")
-  end,
-}
