diff --git a/lua/config/options.lua b/lua/config/options.lua
index e176611..54b2264 100644
--- a/lua/config/options.lua
+++ b/lua/config/options.lua
@@ -13,12 +13,77 @@ local global = vim.g
 local window_opt = vim.wo
 local cmd = vim.cmd
 
+local options = {
+  backup = false,
+  clipboard = "unnamedplus",
+  autoread = true,
+  cmdheight = 0,
+  completeopt = { "menuone", "noselect" },
+  conceallevel = 2,
+  fileencoding = "utf-8",
+  hlsearch = true,
+  incsearch = true,
+  ignorecase = true,
+  mouse = "a",
+  pumheight = 10,
+  showmode = false,
+  showtabline = 2,
+  smartcase = true,
+  smartindent = true,
+  splitbelow = true,
+  splitkeep = vim.fn.has("nvim-0.10.0") == 1 and "screen" or nil,
+  splitright = true,
+  swapfile = false,
+  termguicolors = true,
+  timeout = true,
+  timeoutlen = 400,
+  undofile = true,
+  updatetime = 300,
+  signcolumn = "yes:2",
+  writebackup = false,
+  expandtab = true,
+  softtabstop = 2,
+  shiftwidth = 2,
+  tabstop = 2,
+  nrformats = "alpha",
+  cursorline = false,
+  number = true,
+  relativenumber = true,
+  scrolloff = 8,
+  sidescrolloff = 8,
+  errorbells = false,
+  title = true,
+  winwidth = 10,
+  winminwidth = 10,
+  breakindent = true,
+  linebreak = true,
+  wrap = false,
+  equalalways = false,
+  ruler = false,
+  infercase = true,
+  virtualedit = "block",
+  fillchars = [[eob: ,fold: ,foldopen:,foldsep: ,foldclose:,diff:]],
+  foldlevelstart = 99,
+  foldmethod = "manual",
+  foldnestmax = 10,
+  foldcolumn = "1",
+  foldlevel = 99,
+  foldenable = true,
+}
+for k, v in pairs(options) do
+  vim.opt[k] = v
+end
+
+if vim.fn.has("nvim-0.10.0") == 1 then
+  vim.opt.diffopt:append("linematch:60")
+end
+
 -- This file is automatically loaded by plugins.core
 global.mapleader = ","
-window_opt.relativenumber = false
-window_opt.number = true
-window_opt.numberwidth = 4
-window_opt.signcolumn = "yes:2"
+-- window_opt.relativenumber = false
+-- window_opt.number = true
+-- window_opt.numberwidth = 4
+-- window_opt.signcolumn = "yes:2"
 
 vim.o.termguicolors = true --Set colorscheme (order is important here)
 
