diff --git a/lua/plugins/telescope/undo.lua b/lua/plugins/telescope/undo.lua
index 6ed5b7d..371b85f 100644
--- a/lua/plugins/telescope/undo.lua
+++ b/lua/plugins/telescope/undo.lua
@@ -1,31 +1,32 @@
 return {
-  "debugloop/telescope-undo.nvim",
-  -- commit = "03ff45fab0c4adad4d252e25b5b194e22caf5b4f",
-  dependencies = {
+  {
+    "debugloop/telescope-undo.nvim",
+    -- commit = "03ff45fab0c4adad4d252e25b5b194e22caf5b4f",
+    dependencies = {
       "telescope.nvim",
       "nvim-lua/plenary.nvm",
-  }
-  keys = { { "<leader>su", "<cmd>Telescope undo<cr>", desc = "Undo History" } },
-  config = function()
-    require("telescope").setup({
-      extensions = {
-        undo = {
-          mappings = {
-            i = {
-              ["<cr>"] = require("telescope-undo.actions").yank_additions,
-              ["<S-cr>"] = require("telescope-undo.actions").yank_deletions,
-              ["<C-u>"] = require("telescope-undo.actions").restore,
-            },
-            n = {
-              ["<cr>"] = require("telescope-undo.actions").yank_additions,
-              ["<S-cr>"] = require("telescope-undo.actions").yank_deletions,
-              ["<C-u>"] = require("telescope-undo.actions").restore,
+    },
+    keys = { { "<leader>su", "<cmd>Telescope undo<cr>", desc = "Undo History" } },
+    config = function()
+      require("telescope").setup({
+        extensions = {
+          undo = {
+            mappings = {
+              i = {
+                ["<cr>"] = require("telescope-undo.actions").yank_additions,
+                ["<S-cr>"] = require("telescope-undo.actions").yank_deletions,
+                ["<C-u>"] = require("telescope-undo.actions").restore,
+              },
+              n = {
+                ["<cr>"] = require("telescope-undo.actions").yank_additions,
+                ["<S-cr>"] = require("telescope-undo.actions").yank_deletions,
+                ["<C-u>"] = require("telescope-undo.actions").restore,
+              },
             },
           },
         },
-      },
-    })
-    require("telescope").setup(opts)
-    require("telescope").load_extension("undo")
-  end,
+      })
+      require("telescope").load_extension("undo")
+    end,
+  },
 }
