diff --git a/' b/'
new file mode 100644
index 0000000..e8d0e5f
--- /dev/null
+++ b/'
@@ -0,0 +1,83 @@
+return {
+  "nvim-neorg/neorg",
+  -- dependencies = { "luarocks.nvim" },
+  lazy = false, -- Disable lazy loading as some `lazy.nvim` distributions set `lazy = true` by default
+  -- version = "*", -- Pin Neorg to the latest stable release
+  -- config = true,
+  ft = "norg",
+  cmd = "Neorg",
+  -- build = ":Neorg sync-parsers",
+  config = function()
+    require("neorg").setup({
+      load = {
+        ["core.defaults"] = {},
+        ["core.ui"] = {},
+        ["core.looking-glass"] = {},
+        ["core.keybinds"] = {},
+        ["core.concealer"] = {
+          config = {
+            icon_preset = "diamond",
+            icons = { todo = { done = { icon = "" }, pending = { icon = "" } } },
+          },
+        },
+        ["core.itero"] = {},
+        ["core.esupports.indent"] = {},
+        ["core.summary"] = {},
+        ["core.tempus"] = {},
+        -- ["core.ui.calendar"] = {},
+        ["core.integrations.nvim-cmp"] = {},
+        ["core.integrations.treesitter"] = {},
+        ["core.completion"] = { config = { engine = "nvim-cmp" } },
+        ["core.clipboard"] = {},
+      },
+    })
+  end,
+  --   local opts = {
+  --     load = {
+  --       ["core.defaults"] = {},
+  --       ["core.ui"] = {},
+  --       ["core.looking-glass"] = {},
+  --       ["core.keybinds"] = {},
+  --       ["core.concealer"] = {
+  --         config = {
+  --           icon_preset = "diamond",
+  --           icons = { todo = { done = { icon = "" }, pending = { icon = "" } } },
+  --         },
+  --       },
+  --       ["core.itero"] = {},
+  --       ["core.esupports.indent"] = {},
+  --       ["core.summary"] = {},
+  --       ["core.ui.calendar"] = {},
+  --       ["core.integrations.nvim-cmp"] = {},
+  --       ["core.integrations.treesitter"] = {},
+  --       ["core.completion"] = { config = { engine = "nvim-cmp" } },
+  --       ["core.clipboard"] = {},
+  --     },
+  --   }
+  --   require("neorg").setup(vim.tbl_deep_extend("force", opts, require("plugins.neorg.neorg_workspaces")))
+  -- options for neorg. This will automatically call `require("neorg").setup(opts)`
+  -- opts = {
+  --   load = {
+  --     ["core.defaults"] = {},
+  --     ["core.ui"] = {},
+  --     ["core.looking-glass"] = {},
+  --     ["core.keybinds"] = {},
+  --     ["core.concealer"] = {
+  --       config = {
+  --         icon_preset = "diamond",
+  --         icons = { todo = { done = { icon = "" }, pending = { icon = "" } } },
+  --       },
+  --     },
+  --     ["core.itero"] = {},
+  --     ["core.esupports.indent"] = {},
+  --     ["core.summary"] = {},
+  --     -- ["core.ui.calendar"] = {},
+  --     ["core.integrations.nvim-cmp"] = {},
+  --     ["core.integrations.treesitter"] = {},
+  --     ["core.completion"] = { config = { engine = "nvim-cmp" } },
+  --     ["core.clipboard"] = {},
+  --   },
+  -- },
+
+  -- end,
+}
diff --git a/lua/plugins/luarocks.lua b/lua/plugins/luarocks.lua
new file mode 100644
index 0000000..6bd6f73
--- /dev/null
+++ b/lua/plugins/luarocks.lua
@@ -0,0 +1,5 @@
+return {
+  "vhyrro/luarocks.nvim",
+  priority = 1000,
+  config = true,
+}
diff --git a/lua/plugins/neorg/init.lua b/lua/plugins/neorg/init.lua
index 188668f..bc55423 100644
--- a/lua/plugins/neorg/init.lua
+++ b/lua/plugins/neorg/init.lua
@@ -1,10 +1,14 @@
 return {
   "nvim-neorg/neorg",
+  -- dependencies = { "luarocks.nvim" },
+  lazy = false, -- Disable lazy loading as some `lazy.nvim` distributions set `lazy = true` by default
+  -- version = "*", -- Pin Neorg to the latest stable release
+  -- config = true,
   ft = "norg",
   cmd = "Neorg",
-  build = ":Neorg sync-parsers",
+  -- build = ":Neorg sync-parsers",
   config = function()
-    local opts = {
+    require("neorg").setup({
       load = {
         ["core.defaults"] = {},
         ["core.ui"] = {},
@@ -19,13 +23,61 @@ return {
         ["core.itero"] = {},
         ["core.esupports.indent"] = {},
         ["core.summary"] = {},
+        ["core.tempus"] = {},
         ["core.ui.calendar"] = {},
         ["core.integrations.nvim-cmp"] = {},
         ["core.integrations.treesitter"] = {},
         ["core.completion"] = { config = { engine = "nvim-cmp" } },
         ["core.clipboard"] = {},
       },
-    }
-    require("neorg").setup(vim.tbl_deep_extend("force", opts, require("plugins.neorg.neorg_workspaces")))
+    })
   end,
+  --   local opts = {
+  --     load = {
+  --       ["core.defaults"] = {},
+  --       ["core.ui"] = {},
+  --       ["core.looking-glass"] = {},
+  --       ["core.keybinds"] = {},
+  --       ["core.concealer"] = {
+  --         config = {
+  --           icon_preset = "diamond",
+  --           icons = { todo = { done = { icon = "" }, pending = { icon = "" } } },
+  --         },
+  --       },
+  --       ["core.itero"] = {},
+  --       ["core.esupports.indent"] = {},
+  --       ["core.summary"] = {},
+  --       ["core.ui.calendar"] = {},
+  --       ["core.integrations.nvim-cmp"] = {},
+  --       ["core.integrations.treesitter"] = {},
+  --       ["core.completion"] = { config = { engine = "nvim-cmp" } },
+  --       ["core.clipboard"] = {},
+  --     },
+  --   }
+  --   require("neorg").setup(vim.tbl_deep_extend("force", opts, require("plugins.neorg.neorg_workspaces")))
+  -- options for neorg. This will automatically call `require("neorg").setup(opts)`
+  -- opts = {
+  --   load = {
+  --     ["core.defaults"] = {},
+  --     ["core.ui"] = {},
+  --     ["core.looking-glass"] = {},
+  --     ["core.keybinds"] = {},
+  --     ["core.concealer"] = {
+  --       config = {
+  --         icon_preset = "diamond",
+  --         icons = { todo = { done = { icon = "" }, pending = { icon = "" } } },
+  --       },
+  --     },
+  --     ["core.itero"] = {},
+  --     ["core.esupports.indent"] = {},
+  --     ["core.summary"] = {},
+  --     -- ["core.ui.calendar"] = {},
+  --     ["core.integrations.nvim-cmp"] = {},
+  --     ["core.integrations.treesitter"] = {},
+  --     ["core.completion"] = { config = { engine = "nvim-cmp" } },
+  --     ["core.clipboard"] = {},
+  --   },
+  -- },
+
+  -- end,
 }
